pkg/cvo/internal/operatorstatus.go

```diff
@@ -210,31 +216,40 @@ func waitForOperatorStatusToBeDone(ctx context.Context, interval time.Duration,
 			}
 		}
 
-		condition := failingCondition
-		if degradedCondition != nil {
-			condition = degradedCondition
-		}
-		if condition != nil && condition.Status == configv1.ConditionTrue {
-			message := fmt.Sprintf("Cluster operator %s is reporting a failure", actual.Name)
-			if len(condition.Message) > 0 {
-				message = fmt.Sprintf("Cluster operator %s is reporting a failure: %s", actual.Name, condition.Message)
+		// only available operators need to be checked for degraded condition
+		if available {
+			condition := failingCondition
+			if degradedCondition != nil {
+				condition = degradedCondition
 			}
-			lastErr = &payload.UpdateError{
-				Nested:  errors.New(lowerFirst(message)),
-				Reason:  "ClusterOperatorDegraded",
-				Message: message,
-				Name:    actual.Name,
+			if condition != nil && condition.Status == configv1.ConditionTrue {
+				message := fmt.Sprintf("Cluster operator %s is reporting a failure", actual.Name)
+				if len(condition.Message) > 0 {
+					message = fmt.Sprintf("Cluster operator %s is reporting a failure: %s", actual.Name, condition.Message)
+				}
+				lastErr = &payload.UpdateError{
+					Nested:       errors.New(lowerFirst(message)),
+					UpdateEffect: payload.UpdateEffectNone,
```

Comment:
this is `Available=True, Degraded=True`, right?  Maybe that should be `UpdateEffectFailAfterInterval`?  Possibly by shifting the `updateEffect` definition up before this block and using `updateEffect` here?  I'm personally ok ditching `UpdateEffectFailAfterInterval` and also with not caring about `Degraded`, but while we have both, it seems strange to have `Available=True, Degraded=False, Progressing=True` send us `Failing=True` after the interval, while `Available=True, Degraded=True` stays `Failing=False`.

Reply:
reworked per our discussion